lib/appsignal_phoenix/event_handler.ex

```diff
@@ -92,6 +92,26 @@ defmodule Appsignal.Phoenix.EventHandler do
     @tracer.close_span(@tracer.current_span())
   end
 
+  defp set_span_data(span, %{conn: conn} = metadata) do
+    span
+    |> @span.set_name(name(metadata))
+    |> @span.set_sample_data("params", Appsignal.Metadata.params(conn))
+    |> @span.set_sample_data("environment", Appsignal.Metadata.metadata(conn))
+    |> @span.set_sample_data("session_data", Appsignal.Metadata.session(conn))
+  end
+
+  defp name(%{conn: conn} = metadata) do
+    Appsignal.Metadata.name(conn) || extract_name(metadata)
+  end
+
+  defp extract_name(%{conn: %{method: method}, route: route}) do
+    "#{method} #{route}"
+  end
+
+  defp extract_name(_) do
+    nil
+  end
```

Comment:
We saw [this issue with the Plug instrumentation](https://github.com/appsignal/appsignal-elixir-plug/pull/30) where, if for whatever reason a name cannot be found, the agent will drop the trace, as its root span has no name.

Back then I suggested setting a placeholder name, so that we would at least have some data to debug from. Could that be useful here?

Reply:
The road for that is open again since we’re now (since https://github.com/appsignal/appsignal-elixir-phoenix/issues/57) relying on :telemetry only for the Phoenix integration. This clause was mostly added as a fallback, as we should now catch all plug events. That said, we could add some debug logging here, if needed.

Reply:
Sounds good to me, though I'm not sure what the debug logging would say. There's already debug logging for a span without a name in the agent itself, so merely logging that no name could be extracted does not add more information.

Reply:
Yes, indeed. I think the resulting spans wouldn’t give a lot of information. However, we do have the telemetry event and the conn there, which is what we’d need to debug.